src/lib.rs

```diff
@@ -97,6 +98,65 @@ fn beta(ez: f64) -> f64 {
         + 0.000_424_19 * zl.powi(7)
 }
 
+static INV_POW2: OnceLock<[f64; 64]> = OnceLock::new();
+
+fn inv_pow2_table() -> &'static [f64; 64] {
+    INV_POW2.get_or_init(|| {
+        let mut t = [0.0; 64];
+        let mut val = 1.0f64; // 2^0
+        for slot in t.iter_mut() {
+            *slot = val; // 2^{-k}
+            val *= 0.5; // next
+        }
+        t
+    })
+}
+
+const TQ_USIZE: usize = TQ as usize;
+const TR_USIZE: usize = TR as usize;
+
+// Precomputed ln(1 - b) for the two bin edges per (i, j).
+struct EcTable {
+    ln1p_neg_b1: Vec<Vec<f64>>, // [TQ][TR]
```

Comment:
These `Vec<Vec<f64>>` should be `Box<[[f64; TR as usize]; TQ as usize]>`. This gets rid of like half the opcodes in `Sketch::similarity`.